models/classes/StatisticalMetadata/Import/Processor/ImportProcessor.php

```diff
@@ -0,0 +1,231 @@
+            // @TODO Extract to a separate method
+            try {
+                $this->metadataToBindValidator->validateRelationToResource(
+                    $resource,
+                    $metadataProperties[0][StatisticalMetadataMapper::KEY_DOMAIN]
+                );
+                $this->metadataToBindValidator->validateMetadataValue($record[$metadata]);
+            } catch (InvalidArgumentException $exception) {
+                $reports[] = Report::createWarning(
+                    sprintf(
+                        'Metadata "%s": %s',
+                        $metadata,
+                        $exception->getMessage()
+                    )
+                );
+                ++$errors;
+
+                continue;
+            }
+
+            $metadataToBind[$metadataProperties[0][StatisticalMetadataMapper::KEY_URI]] = $record[$metadata];
+        }
+
+        if ($errors) {
+            // @FIXME Throw an exception instead of error/warning
+            $this->reports[] = $possibleMetadataToBindCount === $errors
+                ? Report::createError(sprintf('Line %d: import failed', $line), null, $reports)
+                : Report::createWarning(sprintf('Line %d: partially imported', $line), null, $reports);
+        }
+
+        return $metadataToBind;
+    }
+
+    /**
+     * @throws tao_models_classes_dataBinding_GenerisInstanceDataBindingException
+     */
+    private function bindProperties(core_kernel_classes_Resource $resource, array $values): void
+    {
+        $binder = new tao_models_classes_dataBinding_GenerisInstanceDataBinder($resource);
+        $binder->bind($values);
+    }
+
+    private function buildFinalReport(int $totalCount, int $updatedResourcesCount): Report
+    {
+        $type = Report::TYPE_SUCCESS;
+        $template = 'CSV import successful: imported %d/%d row(s).';
+        $templateData = [
+            $updatedResourcesCount,
+            $totalCount,
+        ];
+
+        if (!empty($this->reports)) {
+            $type = $updatedResourcesCount === $totalCount
+                ? Report::TYPE_ERROR
+                : Report::TYPE_WARNING;
+            $template = 'Imported %d/%d row(s)';
+
+            [$warningsCount, $errorsCount] = $this->countReportTypes();
+
+            if ($warningsCount) {
+                $template .= '. %d warning(s)';
+                $templateData[] = $warningsCount;
+            }
+
+            if ($errorsCount) {
+                $template .= ', %d error(s)';
+                $templateData[] = $errorsCount;
+            }
+
+            $template .= '.';
+        }
+
+        return new Report($type, vsprintf($template, $templateData), null, $this->reports);
+    }
+
+    private function countReportTypes(): array
+    {
+        $warningsCount = 0;
+        $errorsCount = 0;
+
+        foreach ($this->reports as $report) {
+            $report->getType() === Report::TYPE_WARNING
+                ? ++$warningsCount
+                : ++$errorsCount;
```

Comment:
Suggestion: I'd use a standard `if`/`else` here instead of using a ternary operator with side effects.

```suggestion
            if ($report->getType() === Report::TYPE_WARNING) {
                ++$warningsCount;            
            } else {
                ++$errorsCount;
            }
```

Another approach would be:
```
$warningsCount = array_reduce($this->reports, function ($report, $carry) {
    return $carry + ($report->getType() === Report::TYPE_WARNING ? 1 : 0);
});
$errorCount = count($this->reports) - $warningsCount;
```

(but I think the `if`/`else` is more clear :D)